### PR TITLE
CXP-1357 : Disable Catalogs when Channels associated with mapping are…

### PR DESCRIPTION
**Pull request by hogren**
… removed

**Description (for Contributor and Core Developer)**
image

image

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/src/Infrastructure/Persistence/Catalog/GetCatalogIdsUsingChannelsInMappingQuery.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public function __construct(
+        private readonly Connection $connection,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws \Doctrine\DBAL\Driver\Exception
+     * @throws \Doctrine\DBAL\Exception
+     * @throws \JsonException
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Persistence/Catalog/GetCatalogIdsUsingChannelsInMappingQuery.php:25]: we usually don't add phpdoc for errors that we have no control over or we don't explicitly throw in the function. So I think you can remove these ones.

**Review by masacc — APPROVED** [review]

**Review by guillaumefer-akeneo — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Job/DisableCatalogsOnChannelRemovalTasklet.php [hunk 2]
```diff
@@ -35,7 +37,12 @@ public function execute(): void
         /** @var string[] $channelCodes */
         $channelCodes = $this->stepExecution->getJobParameters()->get('channel_codes');
 
-        $catalogsIds = $this->getCatalogIdsUsingChannelsAsFilterQuery->execute($channelCodes);
+        $catalogsIds = \array_unique(
+            \array_merge(
+                $this->getCatalogIdsUsingChannelsAsFilterQuery->execute($channelCodes),
+                $this->getCatalogIdsUsingChannelsInMappingQuery->execute($channelCodes)
```
Comment (RiffVentura) [inline components/catalogs/back/src/Infrastructure/Job/DisableCatalogsOnChannelRemovalTasklet.php:43]: I'm wondering if it makes sense to combine both of these queries

**Review by RiffVentura — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Job/DisableCatalogsOnChannelRemovalTasklet.php (hunk 2 above)
Comment (hogren) [reply components/catalogs/back/src/Infrastructure/Job/DisableCatalogsOnChannelRemovalTasklet.php:43]: @RiffVentura  You think about create a more big query for the two purposes ?
I asked @masacc about that. The debate is not closed. I choose this solution for the maintainability and reusability. The requests are very small and are executed on very small tables, there is no impacts in performance IMO.

**Review by eoliveauakeneo — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Job/DisableCatalogsOnChannelRemovalTasklet.php (hunk 2 above)
Comment (RiffVentura) [reply components/catalogs/back/src/Infrastructure/Job/DisableCatalogsOnChannelRemovalTasklet.php:43]: Yes, combine both WHERE statements.
But for now, let's keep it simple and maintainable as you did :smile:

**Pull request merged**